### PR TITLE
build: test against published snapshots instead of source deps

### PR DESCRIPTION
**Pull request by raboof**
Refs #2994

### CODE REVIEW

File: project/AkkaDependency.scala [hunk 1]
```diff
@@ -54,15 +55,28 @@ object AkkaDependency {
...
   }
+
+  private def determineLatestSnapshot(prefix: String = ""): String = {
+    import sbt.librarymanagement.Http.http
+    import gigahorse.GigahorseSupport.url
+    import scala.concurrent.Await
+    import scala.concurrent.duration._
+
+    val body = Await.result(http.run(url("https://repo.akka.io/snapshots/com/typesafe/akka/akka-actor_2.12/")), 10.seconds).bodyAsString
+    """href="([^?/].*?)/"""".r.findAllMatchIn(body).map(_.group(1)).filter(_.startsWith(prefix)).toList.last
```
Comment (jrudolph) [inline project/AkkaDependency.scala:80]: Seems a bit brittle but might be good enough. How sure are we that the listing is ordered by version? I.e. the dir listing is ordered lexically and lexical order == version order.

I think we should at least log the result here or somewhere else so it's easier to figure out if somethings goes wrong.

Comment (raboof) [reply project/AkkaDependency.scala:86]: It's ordered by file timestamp right now. That's obviously brittle, though most approaches are... I'd propose using this until we find out we need something better ;).

Comment (jrudolph) [reply project/AkkaDependency.scala:86]: Ok, timestamp sounds good, maybe better than lexical.

File: build.sbt [hunk 2]
```diff
@@ -36,6 +36,10 @@ inThisBuild(Def.settings(
   Formatting.formatSettings,
   shellPrompt := { s => Project.extract(s).currentProject.id + " > " },
   concurrentRestrictions in Global += Tags.limit(Tags.Test, 1),
+  onLoad in Global := {
+    sLog.value.info(s"Building Akka HTTP ${version.value} against Akka ${AkkaDependency.akkaVersion}")
```
Comment (jrudolph) [inline build.sbt:40]: :+1:

**Comment by jrudolph** [comment]
Running a subset of nightlies here: https://jenkins.akka.io:8498/job/akka-http-nightly/

**Comment by jrudolph** [comment]
Looking good (https://jenkins.akka.io:8498/job/akka-http-nightly/AKKA_VERSION=master,Node=akka-http-nightly-node,SCALA_VERSION=2.12.10,jdk=AdoptOpenJDK%208/89/consoleFull):

```
Building Akka HTTP 10.1.11+166-b13ea0be against Akka 2.6.4+117-d611bb0d
```

**Comment by jrudolph** [comment]
I tried reenabling also the builds for `release-2.5` but it failed with a compilation error in the docs. I guess that might be fixed easily. Could you try locally if compiling docs works with `release-2.5`?

**Comment by raboof** [comment]
That's not expected to work since #2996... I guess we should either:

* not build the docs on `release-2.5`
* always build the docs with 2.6.4, even if there's a different version specified for the rest
* Also build the docs with the specified Akka version, unless that's a 2.5 version, in that case fall back to 2.6.4 again

I guess the last is the nicest, even if it's also a bit tricky logic. I'll PR

**Pull request merged**